python/gstaichi/lang/_fast_caching/args_hasher.py

```diff
@@ -79,6 +77,12 @@ def stringify_obj_type(path: tuple[str, ...], obj: Any) -> str | None:
         return "np.bool_"
     if isinstance(obj, enum.Enum):
         return f"enum-{obj.name}-{obj.value}"
+    # The bit in caps should not be modified without updating corresponding test
+    # The rest of free text can be freely modified
+    # (will probably formalize this in more general doc / contributor guidelines at some point)
+    _logging.warn(
+        f"[FASTCACHE][PARAM_INVALID] Parameter with path {path} and type {arg_type} not allowed by fast cache."
+    )
```

Comment:
probably we should type the `obj`  instead of `Any`?

e.g.

```
def stringify_obj_type(path: tuple[str, ...], obj: AllowedObj) -> str | None:
```

Reply:
To do thinks properly, you should probably implement some typing protocol: https://typing.python.org/en/latest/spec/protocol.html#protocol-members

Or move to `pydantic`. It should work in this context.

Still, the most correct weak annotation in this context is `typing.Type`, since you can pass any class.

Reply:
(for example, for fields, this means we can access the snode id)

Reply:
updated to `object`

Reply:
> oh, for some reason I thought you wrote 'type', but now is ays 'object'

Yes, I updated my answer while you were looking somewhere else!